src/bridge/SequencerInbox.sol

```diff
@@ -40,12 +41,16 @@ import "../precompiles/ArbGasInfo.sol";
 import "../precompiles/ArbSys.sol";
 import "../libraries/IReader4844.sol";
 
+import "../data-availability/IAvailDABridge.sol";
+import "../data-availability/MerkleProofInput.sol";
+import "../data-availability/BlobPointer.sol";
```

Comment:
ideally this part should be abstracted away. Based on an input, a contract which is imported instead of these handles the object which is created and manages all operations on that based on a single interface.

This design couples avail tightly with the existing contracts, hence can be tough to be accepted in upstream. If you make a wrapper contract to handle which type of da it is, and then that wrapper contract / module handles everything else, that seperates this from the core logic and hence easier to expand by any third party also + higher acceptance rate due to decoupled nature of the design